com.ekitabu.eKitabu.json

```
@@ -0,0 +1,53 @@
+{
+    "app-id": "com.ekitabu.eKitabu",
+    "base": "io.atom.electron.BaseApp",
+    "base-version": "20.08",
+    "runtime": "org.freedesktop.Platform",
+    "runtime-version": "22.08",
+    "sdk": "org.freedesktop.Sdk",
+    "command": "ekitabu",
+    "separate-locales": false,
+    "finish-args": [
+        "--socket=pulseaudio",
+        "--share=network",
+        "--filesystem=xdg-documents:ro",
+        "--filesystem=xdg-videos:ro",
+        "--filesystem=xdg-download",
+        "--talk-name=com.canonical.AppMenu.Registrar"
+    ],
+    "modules": [
+        {
+            "name": "ekitabu",
+            "buildsystem": "simple",
+            "build-commands": [
+                "install eKitabu_4.5.7_amd64.deb ${FLATPAK_DEST}/bin/ekitabu",
+                "install -Dm644 com.eKitabu.appdata.xml ${FLATPAK_DEST}/share/appdata/${FLATPAK_ID}.appdata.xml",
+                "install -Dm644 eKitabu.png ${FLATPAK_DEST}/share/icons/hicolor/512x512/apps/${FLATPAK_ID}.png"
+            ],
+            "sources": [
+                {
+                    "type": "file",
+                    "path": "com.eKitabu.appdata.xml"
+                },
+                {
+                    "type": "file",
+                    "path": "eKitabu.png"
+                },
+                {
+                    "type": "file",
+                    "filename": "./eKitabu_4.5.7_amd64.deb",
+                    "url": "https://github.com/ekitabu-LLC/flathub/releases/download/v4.0.1/eKitabu_4.5.7_amd64.deb",
+                    "sha256": "1460e1b02d300b3cb98299d407ed9bed5b96357f375e71dacb1ba7300e989202",
+                    "size": 76245298,
+                    "x-checker-data": {
+                        "type": "html",
+                        "url": "https://github.com/ekitabu-LLC/flathub/releases/tag/v4.0.1",
+                        "version-pattern": "Version ([\\d\\.]+)",
+                        "url-template": "https://github.com/ekitabu-LLC/flathub/releases/download/v4.0.1/eKitabu_4.5.7_amd64.deb",
+                        "is-main-source": true
+                    }
```

Because you are asking the checker to check the tag "v4.0.1", so it will only ever detect the same version. You might have better luck with:

```suggestion
                    "x-checker-data": {
                        "type": "json",
                        "url": "https://api.github.com/repos/ekitabu-LLC/flathub/tags",
                        "version-query": ".tag_name | sub(\"^v\"; \"\")",
                        "url-query": "\"https://github.com/ekitabu-LLC/flathub/releases/download/\\(.[0].name)/eKitabu_\\(.[0].name | sub(\"^v\"; \"\"))_amd64.deb\"",
                        "is-main-source": true
                    }
```

(untested)